server/src/main/java/com/cloud/network/NetworkServiceImpl.java

```diff
@@ -1348,11 +1363,54 @@ public Network createGuestNetwork(CreateNetworkCmd cmd) throws InsufficientCapac
         return network;
     }
 
+    /**
+     * Retrieve information (if set) for private VLAN when creating the network
+     */
+    protected Pair<String, PVlanType> getPrivateVlanPair(String pvlanId, String pvlanTypeStr, String vlanId) {
+        String secondaryVlanId = pvlanId;
+        PVlanType type = null;
+
+        if (isNotBlank(secondaryVlanId) && (secondaryVlanId.startsWith("i") || secondaryVlanId.startsWith("c") || secondaryVlanId.startsWith("p"))) {
+            String[] parts = secondaryVlanId.split("-");
+            if (parts.length < 2) {
+                throw new CloudRuntimeException("Could not properly parse the secondary VLAN ID to format TYPE-SVLANID: " + secondaryVlanId);
+            }
+            secondaryVlanId = parts[1];
+            type = PVlanType.fromValue(parts[0]);
```

Comment:
above a check for >2 but below ==2 is expected, potential IndexOutOfRange?
No check for numerical?

Reply:
Not actually IndexOutOfRange as the check is to ensure >= 2, so in worst case at least the == 2 case is covered. Adding numerical check

Reply:
get it. my bad